api/Makefile

```diff
@@ -51,15 +51,15 @@ gen-files .generate_files: lint-cache-dir clean-generated
 	$(DOCKER_RUN) $(CALICO_BUILD) \
 	   sh -c '$(GIT_CONFIG_SSH) defaulter-gen \
 		--v 1 --logtostderr \
-		--go-header-file "/go/src/$(PACKAGE_NAME)/hack/boilerplate/boilerplate.go.txt" \
+		--go-header-file "/go/src/$(PACKAGE_NAME)/../hack/boilerplate/boilerplate.go.txt" \
```

Comment:
We actually need to keep the api files where they are - we mirror the entire contents of this directory to `github.com/projectcalico/api` and so it needs to be able to stand on its own.

I think we should leave `hack/` within this directory and revert these makefile changes.